identity/app/views/consentJourney.scala.html

```diff
@@ -39,51 +39,48 @@ <h3 class="manage-account__switch-title"></h3>
-    @if(step.show) {
-        @(step.banner.map { title =>
-            Html(s"<p class='form__success'>$title</p>")
-        })
-        <fieldset
-            class="identity-wizard__step"
-            data-wizard-step-name="@step.name"
-            role="dialog"
-            aria-labelledby="consentWizard@{step.name.capitalize}Title"
-        >
-            <legend id="consentWizard@{step.name.capitalize}Title" class="identity-title identity-title--page-title">@step.title</legend>
-            <div class="identity-forms-wrapper">
-                @if(step.help || showAgeStep(index)) {
-                    <div class="identity-forms-wrapper__title">
-                        @((step.help ++ showAgeStep(index)).map(paragraph =>
-                            paragraph match {
-                                case m: ConsentStepHelpLegalText => {
-                                    Html(s"<p class='identity-title-explainer identity-title-explainer--small'>${m.text}</p>")
-                                }
-                                case m: ConsentStepHelpText => {
-                                    Html(s"<p class='identity-title-explainer'>${m.text}</p>")
-                                }
+@renderStep(step: ConsentStep, isFirst: Boolean = false, isLast: Boolean = false) = {
+    @(step.banner.map { title =>
+        Html(s"<p class='form__success'>$title</p>")
+    })
+    <fieldset
+        role="dialog"
+        aria-labelledby="consentWizard@{step.name.capitalize}Title"
+    >
+        <legend id="consentWizard@{step.name.capitalize}Title" class="identity-title identity-title--page-title">@step.title</legend>
+        <div class="identity-forms-wrapper">
+            @if(step.help || showAgeStep(isFirst)) {
+                <div class="identity-forms-wrapper__title">
+                    @((step.help ++ showAgeStep(isFirst)).map(paragraph =>
+                        paragraph match {
+                            case m: ConsentStepHelpLegalText => {
+                                Html(s"<p class='identity-title-explainer identity-title-explainer--small'>${m.text}</p>")
                             }
-                        ))
-                    </div>
-                }
-                <div class="identity-forms-wrapper__fields">
-                    @step.content
+                            case m: ConsentStepHelpText => {
+                                Html(s"<p class='identity-title-explainer'>${m.text}</p>")
+                            }
+                        }
+                    ))
                 </div>
+            }
+            <div class="identity-forms-wrapper__fields">
+                @step.content
             </div>
+        </div>
 
-        </fieldset>
-    }
+    </fieldset>
 }
 @renderSteps(steps:List[ConsentStep]) = @{
-    steps.zipWithIndex.map{ case (element, index) => renderStep(element, Some(index)) }
+    val displaySteps = steps.filter(_.show)
```

Comment:
Just inline like so

```
steps
  .filter(_.show)
  .zipWithIndex
  .map(case (element, index) => ...)
```

Reply:
I need to access the size of the filtered `steps`